contracts/modules/maker/MakerV2Invest.sol

```diff
@@ -0,0 +1,178 @@
+        }
+    }
+
+    function dsrBalance(BaseWallet _wallet) public view returns (uint256) {
+        return pot.chi().mul(pot.pie(address(_wallet))) / RAY;
+    }
+
+    /**
+    * @dev lets the owner deposit MCD DAI into the DSR Pot.
+    * @param _wallet The target wallet.
+    * @param _amount The amount of DAI to deposit
+    */
+    function joinDsr(
+        BaseWallet _wallet,
+        uint256 _amount
+    )
+        internal
+    {
+        if (daiToken.balanceOf(address(_wallet)) < _amount) {
+            swapSaiToDai(_wallet, _amount - daiToken.balanceOf(address(_wallet)));
+        }
+
+        // Execute drip to get the chi rate updated to rho == now, otherwise join will fail
+        pot.drip();
+        // Approve DAI adapter to take the DAI amount
+        invokeWallet(address(_wallet), address(daiToken), 0, abi.encodeWithSelector(ERC20_APPROVE, address(daiJoin), _amount));
+        // Join DAI into the vat (_amount of external DAI is burned and the vat transfers _amount of internal DAI from the adapter to the _wallet)
+        invokeWallet(address(_wallet), address(daiJoin), 0, abi.encodeWithSelector(ADAPTER_JOIN, address(_wallet), _amount));
+        // Approve the pot to take out (internal) DAI from the wallet's balance in the vat
+        grantVatAccess(_wallet, address(pot));
+        // Compute the pie value in the pot
+        uint256 pie = _amount.mul(RAY) / pot.chi();
+        // Join the pie value to the pot
+        invokeWallet(address(_wallet), address(pot), 0, abi.encodeWithSelector(POT_JOIN, pie));
+    }
+
+    /**
+    * @dev lets the owner withdraw MCD DAI from the DSR Pot.
+    * @param _wallet The target wallet.
+    * @param _fraction The fraction of invested tokens to exit in per 10000.
+    */
+    function exitDsr(
+        BaseWallet _wallet,
+        uint256 _fraction
+    )
+        internal
+    {
+        // Execute drip to count the savings accumulated until this moment
+        pot.drip();
+        // Calculate the pie wad amount corresponding to the fraction
+        uint256 pie = pot.pie(address(_wallet)).mul(_fraction) / 10000;
+        // Exit DAI from the pot
+        invokeWallet(address(_wallet), address(pot), 0, abi.encodeWithSelector(POT_EXIT, pie));
+        // Allow adapter to access the _wallet's DAI balance in the vat
+        grantVatAccess(_wallet, address(daiJoin));
+        // Check the actual balance of DAI in the vat after the pot exit
+        uint bal = vat.dai(address(_wallet));
+        // It is necessary to check if due to rounding the exact amount can be exited by the adapter.
+        // Otherwise it will do the maximum DAI balance in the vat
+        uint256 amount = pot.chi().mul(pie) / RAY;
+        uint256 withdrawn = bal >= amount.mul(RAY) ? amount : bal / RAY;
+        invokeWallet(address(_wallet), address(daiJoin), 0, abi.encodeWithSelector(ADAPTER_EXIT, address(_wallet), withdrawn));
+    }
+}
```

Comment:
You have removed the `exitAllDsr` which was present in the previous version of the module. Don't the clients need it to remove all collateral?

Reply:
They will be able to use `exitDsr(_wallet, 10000)`, which is equivalent. 

Note that, because on the current `prod` deployment they need to call `exitAllDsr()` (`exitDsr(_wallet, 10000)` would leave some dust in the pot), I've also temporarily deployed a version of MakerV2 on `test` with an `exitAllDsr()` function defined as:
```
    function exitAllDsr(
        BaseWallet _wallet
    )
        external
        onlyWalletOwner(_wallet)
        onlyWhenUnlocked(_wallet)
    {
        exitDsr(_wallet, 10000);
    }
```
The idea is to change the clients from calling `exitAllDsr()` to calling `exitDsr(10000)` after we are ready to the deploy CDP-powered `MakerV2Manager` (in which `exitDsr(10000)` no longer leaves dust in the pot).